framework/test/unit/node/consensus/certificate_generation/utils.spec.ts

```diff
@@ -12,16 +12,94 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
+import {
+	signBLS,
+	generatePrivateKey,
+	getPublicKeyFromPrivateKey,
+	getRandomBytes,
+} from '@liskhq/lisk-cryptography';
+import { codec } from '@liskhq/lisk-codec';
+import { Certificate } from '../../../../../src/node/consensus/certificate_generation/types';
+import { MESSAGE_TAG_CERTIFICATE } from '../../../../../src/node/consensus/certificate_generation/constants';
+import { certificateSchema } from '../../../../../src/node/consensus/certificate_generation/schema';
+import { verifySingleCertificateSignature } from '../../../../../src/node/consensus/certificate_generation/utils';
+
 // eslint-disable-next-line jest/no-disabled-tests
-describe.skip('utils', () => {
+describe('utils', () => {
+	const networkIdentifier = Buffer.alloc(0);
+
 	describe('computeCertificateFromBlockHeader', () => {
 		it.todo('');
 	});
 	describe('signCertificate', () => {
 		it.todo('');
 	});
 	describe('verifySingleCertificateSignature', () => {
-		it.todo('');
+		let privateKey: Buffer;
+		let publicKey: Buffer;
+		let certificate: Certificate;
+		let signature: Buffer;
+
+		beforeEach(() => {
+			privateKey = generatePrivateKey(getRandomBytes(32));
+			publicKey = getPublicKeyFromPrivateKey(privateKey);
+			certificate = {
+				blockID: Buffer.alloc(0),
+				height: 1030,
+				stateRoot: Buffer.alloc(0),
+				timestamp: 10300,
+				validatorsHash: Buffer.alloc(0),
+			};
+
+			const encodedCertificate = codec.encode(certificateSchema, certificate);
+
+			signature = signBLS(
+				MESSAGE_TAG_CERTIFICATE,
+				networkIdentifier,
+				encodedCertificate,
+				privateKey,
+			);
+
+			(certificate as any).aggregationBits = Buffer.alloc(0);
+			(certificate as any).signature = Buffer.alloc(0);
```

Comment:
maybe better to assign actual  bytes to see  if it's ignored

Reply:
done